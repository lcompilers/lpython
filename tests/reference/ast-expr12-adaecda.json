{
    "basename": "ast-expr12-adaecda",
    "cmd": "lpython --show-ast --no-color {infile} -o {outfile}",
    "infile": "tests/expr12.py",
    "infile_hash": "bad9ac6e0956fddb636f4e806f4a97c27396adc2416640bd291d2dc8",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast-expr12-adaecda.stdout",
    "stdout_hash": "b4fdb14f66196f4692126efb3b349707d3f862d6c045b61125af41cb",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}