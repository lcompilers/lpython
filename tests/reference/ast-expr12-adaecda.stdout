(Module
    [(FunctionDef
        test
        ([]
        [(a
        (Name
            i32
            Load
        )
        ())
        (b
        (Name
            i32
            Load
        )
        ())]
        []
        []
        []
        []
        [])
        [(Return
            (BinOp
                (Name
                    a
                    Load
                )
                Pow
                (Name
                    b
                    Load
                )
            )
        )]
        []
        (Name
            i32
            Load
        )
        ()
    )
    (FunctionDef
        check
        ([]
        []
        []
        []
        []
        []
        [])
        [(AnnAssign
            (Name
                a
                Store
            )
            (Name
                i32
                Load
            )
            ()
            1
        )
        (Assign
            [(Name
                a
                Store
            )]
            (Call
                (Name
                    test
                    Load
                )
                [(ConstantInt
                    2
                    ()
                )
                (ConstantInt
                    2
                    ()
                )]
                []
            )
            ()
        )
        (Return
            (Name
                a
                Load
            )
        )]
        []
        (Name
            i32
            Load
        )
        ()
    )
    (FunctionDef
        main0
        ([]
        []
        []
        []
        []
        []
        [])
        [(AnnAssign
            (Name
                x
                Store
            )
            (Name
                i32
                Load
            )
            ()
            1
        )
        (Assign
            [(Name
                x
                Store
            )]
            (Call
                (Name
                    check
                    Load
                )
                []
                []
            )
            ()
        )]
        []
        ()
        ()
    )
    (Expr
        (Call
            (Name
                main0
                Load
            )
            []
            []
        )
    )]
    []
)
